resolver/src/p2p/ingress.rs

```diff
@@ -88,3 +92,28 @@ impl<K: Span> Resolver for Mailbox<K> {
             .expect("Failed to send cancel_all");
     }
 }
+
+impl<K: Span, P: PublicKey> Mailbox<K, P> {
+    /// Register a peer as likely having data for a key.
+    ///
+    /// When fetching this key, hinted peers are tried first. If a hinted peer
+    /// fails (timeout, error response, or send failure), only that peer is
+    /// removed from hints. As hints deplete through failures, the resolver
+    /// naturally falls back to trying all peers.
+    ///
+    /// Multiple hints can be registered for the same key. Hints can be added
+    /// before or after calling `fetch()` - new hints will be used on retry.
+    ///
+    /// Hints are automatically cleared when:
+    /// - The fetch succeeds (data received)
+    /// - The fetch is canceled
+    /// - All hinted peers fail (empty hints trigger fallback to all peers)
+    ///
+    /// Panics if the send fails.
+    pub async fn hint(&mut self, key: K, peer: P) {
```

Comment:
Should this be a separate operation than enqueueing the fetch? This seems error prone in the case where someone mistakenly hints following a fetch and that fetch is already resolved, for example - the hint would then sit in memory forever.

Instead maybe fetch should take an optional hint of peers that might have the data along with the list of keys, so that we can prevent that misuse?

Reply:
(or, we could add a TTL / ring buffer for hints, though I like combining the hint w/ the fetch a bit more.)

Reply:
Good point. I implemented it this way as I was targeting the use case in #2287, i.e. we trigger a fetch when we see some peer in a future epoch, but we also keep adding other peers as hints as we see them. In #2287 I just called `resolver.clear()` once we got the finalization, which clears all the hints. But I agree that the API might be prone to misuse. I think being able to add hints after the fetch has started is useful, so maybe the API can be as follows:
- add `fetch_hinted` that takes a vec of peers as hints
- `hint` only adds the hint if there is any on-going request for that key, otherwise the hint is ignored